packages/packages/v1-adapters/src/dispatchers/dispatchers.ts

```diff
@@ -0,0 +1,38 @@
+import { ExtendedWindow } from './types';
+import { isJQueryDeferred, promisifyJQueryDeferred } from './utils';
+
+export function runCommand( command: string, args?: object ) : Promise<unknown> {
+	const extendedWindow = window as unknown as ExtendedWindow;
+
+	if ( ! extendedWindow.$e?.run ) {
+		return Promise.reject( '`$e.run()` is not available' );
+	}
+
+	const result = extendedWindow.$e.run( command, args );
+
+	if ( result instanceof Promise ) {
+		return result;
+	}
+
+	if ( isJQueryDeferred( result ) ) {
+		return promisifyJQueryDeferred( result );
+	}
+
+	return Promise.resolve( result );
+}
+
+export function openRoute( route: string ) : Promise<void> {
+	const extendedWindow = window as unknown as ExtendedWindow;
```

Comment:
why twice? why not at the top of the module?

Reply:
IDK, maybe `window` isn't fully initialized yet when the module loads, some references might change, didn't want to take that risk 🤷